src/units/abilities.cpp

```diff
@@ -1111,11 +1111,33 @@ namespace { // Helpers for attack_type::special_active()
 
 unit_ability_list attack_type::list_ability(const std::string& ability) const
 {
+	const unit_map& units = display::get_singleton()->get_units();
+
+	unit_const_ptr self = self_;
+	unit_const_ptr other = other_;
```

Comment:
Is there a reason to copy these pointers? If it's just because you don't like the trailing underscore, then I'd say that's not a good enough reason.

Reply:
@CelticMinstrel it is because https://github.com/wesnoth/wesnoth/commit/f12d51a2f9af13e5edd4c44a32b2ee8c0ea15862 of @gfgtdf , but i don't know if is fix list_ability.

Reply:
it is fixed

Reply:
the copy of pointer was for is compatibility with https://github.com/wesnoth/wesnoth/pull/5016/commits/589780410797425176719572bbc85ce7d09b0698